packages/ubuntu_provision/lib/src/keyboard/keyboard_page.dart

```diff
@@ -15,85 +22,233 @@ class KeyboardPage extends ConsumerWidget with ProvisioningPage {
+                  itemBuilder: (index) => model.variantName(index),
+                  label: 'Keyboard variant',
+                  onChanged: (index) {
+                    if (index != null) {
+                      model.selectVariant(index);
+                      DebouncedAnnouncer.announce('Selected variant ${model.variantName(index)}');
+                    }
+                  },
+                ),
+                ),
               ),
-            ),
-          ],
+            ],
+          ),
         ),
+        
         const SizedBox(height: kWizardSpacing),
         const Divider(height: 1),
         const SizedBox(height: kWizardSpacing),
-        TextField(
-          decoration: InputDecoration(
-            hintText: lang.keyboardTestHint,
+        
+        // 3. Test Field with accessibility
+        Semantics(
+          textField: true,
+          label: 'Keyboard test field ${lang.keyboardTestHint}',
+          child: Focus(
+            focusNode: _testFieldFocusNode,
+            child: TextField(
+              decoration: InputDecoration(
+                hintText: lang.keyboardTestHint,
+              ),
+            ),
           ),
         ),
       ],
     );
   }
 }
 
-final _nextFocusNodeProvider = ProvisioningPage.createNextFocusNodeProvider();
+
+
+// Simplified accessible list widget
+class AccessibleListWidget extends StatelessWidget {
+  final int selectedIndex;
+  final int itemCount;
+  final Widget Function(BuildContext, int) itemBuilder;
+  final Function(int) onSelectionChanged;
+  final FocusNode nextFocusNode;
+  final Function(String)? onSearch;
+
+  const AccessibleListWidget({
+    Key? key,
+    required this.selectedIndex,
+    required this.itemCount,
+    required this.itemBuilder,
+    required this.onSelectionChanged,
+    required this.nextFocusNode,
+    this.onSearch,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return KeyboardListener(
+      focusNode: FocusNode(),
+      onKeyEvent: (KeyEvent event) {
+        if (event is KeyDownEvent) {
+          if (event.logicalKey == LogicalKeyboardKey.arrowDown) {
+            final newIndex = (selectedIndex + 1) % itemCount;
+            onSelectionChanged(newIndex);
+          } else if (event.logicalKey == LogicalKeyboardKey.arrowUp) {
+            final newIndex = (selectedIndex - 1 + itemCount) % itemCount;
+            onSelectionChanged(newIndex);
+          } else if (event.logicalKey == LogicalKeyboardKey.tab && !HardwareKeyboard.instance.isShiftPressed) {
+            nextFocusNode.requestFocus();
+          }
+        }
+      },
+      child: ListWidget.builder(
+        selectedIndex: selectedIndex,
+        itemCount: itemCount,
+        tabFocusNode: nextFocusNode,
+        itemBuilder: itemBuilder,
+        onKeySearch: onSearch,
+      ),
+    );
+  }
+}
+
+final _nextFocusNodeProvider = ProvisioningPage.createNextFocusNodeProvider();
```

Comment:
File-global variables like this should be near the top of the file instead of the bottom.